addons/pinniped/config-controller/controllers/v1_controller_test.go

```diff
@@ -0,0 +1,232 @@
+			cluster2 := &clusterapiv1beta1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: pinnipedNamespace,
+					Name:      "another-name",
+					Labels: map[string]string{
+						tkrLabel: testTKRLabel,
+					},
+				},
+				Spec: clusterapiv1beta1.ClusterSpec{
+					InfrastructureRef: &corev1.ObjectReference{
+						Kind: tkgconstants.InfrastructureRefVSphere,
+						Name: "another-name",
+					},
+				},
+			}
+
+			clusters = []*clusterapiv1beta1.Cluster{cluster, cluster2}
+
+			for _, c := range clusters {
+				createObject(ctx, c)
+			}
+		})
+
+		AfterEach(func() {
+			deleteObject(ctx, configMap)
+
+			for _, c := range clusters {
+				deleteObject(ctx, c)
+			}
+		})
+		When("the configmap gets created", func() {
+			BeforeEach(func() {
+				createObject(ctx, configMap)
+			})
+
+			It("creates all the addons secrets", func() {
+				for _, c := range clusters {
+					Eventually(verifySecretFunc(ctx, c, configMap, true)).Should(Succeed())
+				}
+			})
+		})
+
+		When("the configmap gets updated", func() {
+			var configMapCopy *corev1.ConfigMap
+			BeforeEach(func() {
+				createObject(ctx, configMap)
+				configMapCopy = configMap.DeepCopy()
+				configMapCopy.Data[issuerKey] = issuer
+				configMapCopy.Data[issuerCABundleKey] = issuerCABundleData
+				err := k8sClient.Update(ctx, configMapCopy)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(func(g Gomega) {
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(configMapCopy), configMapCopy)
+					g.Expect(err).NotTo(HaveOccurred())
+				}).Should(Succeed())
+			})
+
+			It("updates all the addon secrets", func() {
+				for _, c := range clusters {
+					Eventually(verifySecretFunc(ctx, c, configMapCopy, true)).Should(Succeed())
+				}
+			})
+		})
+
+		When("the configmap does not have an issuer or caBundle", func() {
+			var configMapCopy *corev1.ConfigMap
+			BeforeEach(func() {
+				configMapCopy = configMap.DeepCopy()
+				configMapCopy.Data["fake_data"] = "something-fake"
+				delete(configMapCopy.Data, issuerKey)
+				delete(configMapCopy.Data, issuerCABundleKey)
+				err := k8sClient.Create(ctx, configMapCopy)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(func(g Gomega) {
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(configMapCopy), configMapCopy)
+					g.Expect(err).NotTo(HaveOccurred())
+				}).Should(Succeed())
+			})
+
+			AfterEach(func() {
+				deleteObject(ctx, configMapCopy)
+			})
+
+			It("passes through an empty string for the value", func() {
+				// TODO: what should happen in this scenario?
+				for _, c := range clusters {
+					Eventually(verifySecretFunc(ctx, c, configMapCopy, true)).Should(Succeed())
+				}
+			})
+		})
+
+		When("a configmap in a different namespace gets created", func() {
+			var configMapCopy *corev1.ConfigMap
+			BeforeEach(func() {
+				createObject(ctx, configMap)
+				configMapCopy = configMap.DeepCopy()
+				configMapCopy.Namespace = pinnipedNamespace
+				configMapCopy.Data["issuer"] = issuer
```

Comment:
i think even if the controller got triggered by this `configMapCopy`, it would produce the same secret that gets created by `configMap` here. might we change the `configMapCopy.Data["issuer"]` and `configMapCopy.Data["issuer_ca_bundle_data"]` to something different than `issuer` and `issuerCABundleData` to make sure `configMapCopy` _really_ doesn't get used?

Reply:
So in the original configMap, the issuer and ca bundle get set to the following:
```
					"issuer":                "tuna.io",
					"issuer_ca_bundle_data": "ball-of-fluff",
```

we are using the following vars:
```
			issuer             = "cats.dev"
			issuerCABundleData = "secret-blanket"
```

...so they actually are utilizing different values.... I think you were the one who set those `conts` at some point during a refactor.... likely `go` prompted you.  I could make all of these config map copies utilize different values instead of using those `conts` if that is less confusing.

Reply:
I ended up just editing all of them to utilize totally different things and took out the constant values!